// ap-angular2-fullcalendar/src/calendar/calendar.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var import0 = require("@angular/core");
var import1 = require("./calendar");
var styles_CalendarComponent = [];
exports.RenderType_CalendarComponent = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_CalendarComponent,
    data: {}
});
function View_CalendarComponent_0(l) {
    return import0.ɵvid(0, [(l()(), import0.ɵeld(0, null, null, 0, 'div', [], null, null, null, null, null))], null, null);
}
exports.View_CalendarComponent_0 = View_CalendarComponent_0;
function View_CalendarComponent_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'angular2-fullcalendar', [], null, null, null, View_CalendarComponent_0, exports.RenderType_CalendarComponent)),
        import0.ɵdid(4243456, null, 0, import1.CalendarComponent, [import0.ElementRef], null, null)
    ], null, null);
}
exports.CalendarComponentNgFactory = import0.ɵccf('angular2-fullcalendar', import1.CalendarComponent, View_CalendarComponent_Host_0, { options: 'options' }, { initialized: 'initialized' }, []);
